### PR TITLE
Removing circom

### PR DESCRIPTION
**Pull request by mrsmkl**
## Description

The parts that depended on circom are now in another repo.

## Is this PR related with an open issue?

Related to Issue #

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
 
## Checklist:

- [ ] Follows the code style of this project.
- [ ] Tests Cover Changes
- [ ] Documentation

#### Funny gif

Put a link of a funny gif inside the parenthesis-->

### CODE REVIEW

File: integration/nevermined/ConsumeAsset.test.ts [hunk 1]
```diff
@@ -104,8 +105,9 @@ describe('Consume Asset', () => {
 
     it('should get the agreement conditions status not fulfilled', async () => {
         // todo change this, a test should never dependent on the previous test because the order might change during runtime
-        await new Promise(resolve => setTimeout(resolve, 500))
-        const status = await nevermined.agreements.status(agreementId)
+        await new Promise(resolve => setTimeout(resolve, 3000))
+        const status = await repeat(3, nevermined.agreements.status(agreementId))
+        // const status = await nevermined.agreements.status(agreementId)
```
Comment (r-marques) [inline integration/nevermined/ConsumeAsset.test.ts:110]: cleanup

File: integration/nevermined/ConsumeAssetGateway.test.ts [hunk 2]
```diff
@@ -88,8 +89,8 @@ describe('Consume Asset (Gateway)', () => {
 
     it('should get the lockPayment condition fulfilled', async () => {
         // todo change this, a test should never dependent on the previous test because the order might change during runtime
-        await new Promise(resolve => setTimeout(resolve, 500))
-        const status = await nevermined.agreements.status(agreementId)
+        await new Promise(resolve => setTimeout(resolve, 3000))
```
Comment (r-marques) [inline integration/nevermined/ConsumeAssetGateway.test.ts:92]: we have a sleep function under `integration/utils`

File: integration/nevermined/ConsumeExternalDDOAsset.test.ts [hunk 3]
```diff
@@ -102,8 +103,8 @@ describe('Consume Asset (Documentation example)', () => {
 
     it('should get the agreement conditions status not fulfilled', async () => {
         // todo change this, a test should never dependent on the previous test because the order might change during runtime
-        await new Promise(resolve => setTimeout(resolve, 500))
-        const status = await nevermined.agreements.status(agreementId)
+        await new Promise(resolve => setTimeout(resolve, 3000))
```
Comment (r-marques) [inline integration/nevermined/ConsumeExternalDDOAsset.test.ts:106]: we have a sleep function under `integration/utils`

File: integration/nevermined/SearchAsset.test.ts [hunk 4]
```diff
@@ -72,12 +72,13 @@ describe('Search Asset', () => {
             await nevermined.assets.create(metadataGenerator('Test3', userId), publisher),
             DDO
         )
+        await new Promise(r => setTimeout(r, 1000))
```
Comment (r-marques) [inline integration/nevermined/SearchAsset.test.ts:75]: we have a sleep function under `integration/utils`

File: src/keeper/Keeper.ts [hunk 5]
```diff
@@ -137,7 +132,7 @@ export class Keeper extends Instantiable {
                 ),
                 // Templates
                 accessTemplate: AccessTemplate.getInstance(this.instanceConfig),
-                accessProofTemplate: AccessProofTemplate.getInstance(this.instanceConfig),
+                // accessProofTemplate: AccessProofTemplate.getInstance(this.instanceConfig),
```
Comment (r-marques) [inline src/keeper/Keeper.ts:135]: we have a sleep function under `integration/utils`

File: src/keeper/Keeper.ts [hunk 6]
```diff
@@ -235,7 +229,7 @@ export class Keeper extends Instantiable {
             lockPaymentCondition: this.instances.lockPaymentCondition,
             escrowPaymentCondition: this.instances.escrowPaymentCondition,
             accessCondition: this.instances.accessCondition,
-            accessProofCondition: this.instances.accessProofCondition,
+            // accessProofCondition: this.instances.accessProofCondition,
```
Comment (r-marques) [inline src/keeper/Keeper.ts:232]: cleanup

File: src/keeper/Keeper.ts [hunk 7]
```diff
@@ -328,7 +328,7 @@ export class Keeper extends Instantiable {
         lockPaymentCondition: LockPaymentCondition
         escrowPaymentCondition: EscrowPaymentCondition
         accessCondition: AccessCondition
-        accessProofCondition: AccessProofCondition
+        // accessProofCondition: AccessProofCondition
```
Comment (r-marques) [inline src/keeper/Keeper.ts:331]: cleanup

File: src/keeper/Keeper.ts [hunk 8]
```diff
@@ -345,12 +345,15 @@ export class Keeper extends Instantiable {
...
+    public conditionsList: ConditionSmall[]
+    public templateList: GenericAccess[]
+
     /**
      * Templates instances.
      */
     public templates: {
         accessTemplate: AccessTemplate
-        accessProofTemplate: AccessProofTemplate
+        // accessProofTemplate: AccessProofTemplate
```
Comment (r-marques) [inline src/keeper/Keeper.ts:356]: cleanup

File: src/keeper/contracts/templates/index.ts [hunk 9]
```diff
@@ -1,7 +1,7 @@
 export * from './AgreementTemplate.abstract'
 export { BaseTemplate } from './BaseTemplate.abstract'
 export { AccessTemplate } from './AccessTemplate'
-export { AccessProofTemplate } from './AccessProofTemplate'
+// export { AccessProofTemplate } from './AccessProofTemplate'
```
Comment (r-marques) [inline src/keeper/contracts/templates/index.ts:4]: cleanup

File: src/nevermined/Account.ts [hunk 10]
```diff
@@ -2,7 +2,7 @@ import BigNumber from 'bignumber.js'
 import Balance from '../models/Balance'
 
 import { Instantiable, InstantiableConfig } from '../Instantiable.abstract'
-import { makeKeyTransfer } from '../utils/KeyTransfer'
+// import { makeKeyTransfer } from '../utils/KeyTransfer'
```
Comment (r-marques) [inline src/nevermined/Account.ts:5]: cleanup

File: src/nevermined/Account.ts [hunk 11]
```diff
@@ -37,10 +37,11 @@ export default class Account extends Instantiable {
         return this.babyX.substr(2) + this.babyY.substr(2)
     }
 
+    /*
     public async signBabyjub(num: bigint) {
         const keytransfer = await makeKeyTransfer()
         return keytransfer.signBabyjub(this.babySecret, num)
-    }
+    }*/
```
Comment (r-marques) [inline src/nevermined/Account.ts:44]: cleanup

File: src/nevermined/Assets.ts [hunk 12]
```diff
@@ -945,17 +950,14 @@ export class Assets extends Instantiable {
...
-                this.logger.log('Access proof service Added')
-                await ddo.addService(
-                    this.nevermined,
-                    this.createAccessProofService(
-                        templates,
-                        publisher,
-                        metadata,
-                        accessProofServiceAgreementTemplate
+            for (const name of serviceTypes) {
+                const plugin = this.servicePlugin[name]
```
Comment (r-marques) [inline src/nevermined/Assets.ts:954]: I think you should be able to use `this.servicePlugin.name` in this case

File: src/sdk.ts [hunk 13]
```diff
@@ -37,6 +37,7 @@ export {
 } from './permissions/Permissions.interfaces'
 
 import * as events from './events'
+import * as KeyTransfer from './models/KeyTransfer'
```
Comment (r-marques) [inline src/sdk.ts:40]: do we still need this?

File: src/sdk.ts [hunk 14]
```diff
@@ -48,6 +49,7 @@ export {
     DID,
     Logger,
     Keeper,
+    KeyTransfer,
```
Comment (r-marques) [inline src/sdk.ts:52]: do we still need this?

File: test/nevermined/Account.test.ts [hunk 15]
```diff
@@ -39,7 +39,7 @@ describe('Account', () => {
         })
     })
 
-    describe('#getEthBalance()', () => {
+    describe.skip('#getEthBalance()', () => {
```
Comment (r-marques) [inline test/nevermined/Account.test.ts:42]: any reason to skip this tests?

File: test/nevermined/Account.test.ts [hunk 16]
```diff
@@ -56,7 +56,7 @@ describe('Account', () => {
         })
     })
 
-    describe('#getBalance()', () => {
+    describe.skip('#getBalance()', () => {
```
Comment (r-marques) [inline test/nevermined/Account.test.ts:59]: skip?

File: test/utils/KeyTransfer.test.ts [hunk 17]
```diff
@@ -1,3 +1,4 @@
+/*
```
Comment (r-marques) [inline test/utils/KeyTransfer.test.ts:1]: we can probably delete this file

File: src/sdk.ts (hunk 13 above)
Comment (mrsmkl) [reply src/sdk.ts:40]: yes...

**Review by r-marques — APPROVED** [review]
lgtm

**Pull request merged**